chunks/chunks.go

```diff
@@ -55,8 +55,9 @@ type Meta struct {
 }
 
 // writeHash writes the chunk encoding and raw data into the provided hash.
-func (cm *Meta) writeHash(h hash.Hash) error {
-	if _, err := h.Write([]byte{byte(cm.Chunk.Encoding())}); err != nil {
+func (cm *Meta) writeHash(h hash.Hash, buf []byte) error {
```

Comment:
worth to add a comment about `buf`? Also wonder if it's not worth to hide this buffer in `cm`? especially that it is private function?

Reply:
Will add a comment.

> Also wonder if it's not worth to hide this buffer in cm?

https://github.com/prometheus/tsdb/blob/eb79899bfc3c2edb40ad1ba3af00056d1ecd735e/chunks/chunks.go#L305-L330
That would mean we will duplicate/share those bytes among all the chunk metas (`writeHash` is being called in a loop on different `Meta`). So no, it's not worth.